src/frontend/src/optimizer/property/func_dep.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use fixedbitset::FixedBitSet;
+use itertools::Itertools;
+
+/// [`FunctionalDependency`] represent a dependency of from --> to.
```

Comment:
Maybe add an example to make it more clear. e.g., "For columns ABCD, the FD AC->B is represented as 1010->0100"

Reply:
Maybe also mention here order doesn't matter, or it's combination of columns